src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -56,6 +56,7 @@ public class AvaloniaObject : IAvaloniaObject, IAvaloniaObjectDebug, INotifyProp
         /// </summary>
         public AvaloniaObject()
         {
+            CheckAccess();
```

Comment:
Shouldn't this be `VerifyAccess` instead of `CheckAccess`? Also, do we need this check in the first place?

Reply:
Yeah, that should be VerifyAccess. And yes, we need to ensure that `AvaloniaObject` can be created only from UI thread, there will be hard-to-debug issues otherwise.